service/application/src/test/java/com/solace/maas/ep/event/management/agent/commandManager/CloudManagedEMACommandManagerTests.java

```diff
@@ -0,0 +1,119 @@
+package com.solace.maas.ep.event.management.agent.commandManager;
+
+import com.solace.maas.ep.common.messages.CommandMessage;
+import com.solace.maas.ep.event.management.agent.command.CommandManager;
+import com.solace.maas.ep.event.management.agent.config.eventPortal.EventPortalProperties;
+import com.solace.maas.ep.event.management.agent.plugin.command.model.Command;
+import com.solace.maas.ep.event.management.agent.plugin.command.model.JobStatus;
+import com.solace.maas.ep.event.management.agent.plugin.service.MessagingServiceDelegateService;
+import com.solace.maas.ep.event.management.agent.plugin.solace.processor.semp.SempClient;
+import com.solace.maas.ep.event.management.agent.plugin.solace.processor.semp.SolaceHttpSemp;
+import com.solace.maas.ep.event.management.agent.plugin.terraform.manager.TerraformManager;
+import com.solace.maas.ep.event.management.agent.publisher.CommandPublisher;
+import lombok.extern.slf4j.Slf4j;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.mockito.ArgumentCaptor;
+import org.mockito.stubbing.Answer;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.ActiveProfiles;
+
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static org.awaitility.Awaitility.await;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.annotation.DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD;
+
+@Slf4j
+@ActiveProfiles("TEST")
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT, properties = {
+        "eventPortal.gateway.messaging.standalone=false",
+        "eventPortal.managed=true",
+        "eventPortal.incomingRequestQueueName = ep_core_ema_requests_123456_123123",
+        "event-portal.gateway.messaging.rto-session=false"
+})
+@DirtiesContext(classMode = BEFORE_EACH_TEST_METHOD)
+//CPD-OFF
+class CloudManagedEMACommandManagerTests {
+
```

Comment:
This needs to be a separate class as we want to load a new application context in  cloud managed mode